scopes/component/ui/compare/component-compare.tsx

```diff
@@ -57,16 +58,27 @@ export function ComponentCompare({ navSlot, host, routeSlot }: ComponentCompareP
   const { component: base, loading } = useComponent(host, baseId.toString());
 
   const nothingToCompare = !loading && !compareIsLocalChanges && (component.logs?.length || []) < 2;
+  const showSubMenus = !loading && !nothingToCompare;
+
+  const baseVersionInfo = useMemo(() => allVersionInfo.find(findMatchingVersionInfo(baseId.version)), [baseId]);
+  const compareVersionInfo = useMemo(
+    () => allVersionInfo.find(findMatchingVersionInfo(compare.version)),
+    [compare.version]
+  );
 
   const componentCompareModel: ComponentCompareModel = {
-    compare,
-    base,
+    compare: {
+      model: compare,
+      isLocalChanges: compareIsLocalChanges,
+      versionInfo: compareVersionInfo,
+    },
+    base: base && {
+      model: base,
+      versionInfo: baseVersionInfo,
+    },
```

Comment:
its kinda hard to track what is happening here. not sure I understand why you are doing all this.
lets talk about it